### PR TITLE
React carousel component

### PR DESCRIPTION
**Pull request by edtan094**
react-carousel-component

### CODE REVIEW

File: react-carousel-component/src/carousel.jsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    };
+    this.moveToTheLeft = this.moveToTheLeft.bind(this);
+    this.moveToTheRight = this.moveToTheRight.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.highlightDot = this.highlightDot.bind(this);
+  }
+
+  moveToTheRight() {
+    index++;
+    if (index > 4) {
```
Comment (thebearingedge) [inline react-carousel-component/src/carousel.jsx]: We don't know how many images the carousel will deal with so you'll have to remove any hard-coded numbers like this.

File: react-carousel-component/src/carousel.jsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      index = 0;
+    }
+    this.setState(
+      { image: pokemon[index].image }
+    );
+    this.highlightDot();
+    clearInterval(this.interval);
+    this.componentDidMount();
```
Comment (thebearingedge) [inline react-carousel-component/src/carousel.jsx]: Generally we never call lifecycle methods like this. Instead define a separate custom method that you can call from within here and componentDidMount

File: react-carousel-component/src/carousel.jsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+  highlightDot() {
+    var $circleButtons = document.querySelectorAll('.circle-button');
+    for (let i = 0; i < $circleButtons.length; i++) {
+      if (i === index) {
+        $circleButtons[i].classList.replace('far', 'fas');
+      }
+      if (i !== index) {
+        $circleButtons[i].classList.replace('fas', 'far');
+      }
```
Comment (thebearingedge) [inline react-carousel-component/src/carousel.jsx]: We don't manipulate the DOM like this in React. Instead, pass the correct props to your React elements in `render`

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Ok, good, but we need to reorganize some stuff to make this more React-y.

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Ok, fantastic, but the list of images should be passed to the component in `index.jsx` as a prop.

**Comment by edtan094** [comment]
react-carousel-updated

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**